shell/browser/browser_win.cc

```diff
@@ -169,6 +231,91 @@ void Browser::Focus(gin_helper::Arguments* args) {
   EnumWindows(&WindowsEnumerationHandler, reinterpret_cast<LPARAM>(&pid));
 }
 
+void GetFileIcon(const base::FilePath& path,
+                 v8::Isolate* isolate,
+                 base::CancelableTaskTracker* cancelable_task_tracker_,
+                 const base::FilePath& app_path,
+                 base::string16 app_display_name,
+                 gin_helper::Promise<gin_helper::Dictionary> promise) {
+  base::FilePath normalized_path = path.NormalizePathSeparators();
+  gin_helper::Dictionary dict = gin::Dictionary::CreateEmpty(isolate);
```

Comment:
We only need to construct this dict if `LookupIconFromFilepath` returns.  We can move this allocation to inside the `if (icon) {` block